.github/workflows/snyk-scan.yml

```diff
@@ -0,0 +1,133 @@
+name: Snyk Security Vulnerability Scan
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+  push:
+    tags:
+      - 'v.[0-9]+.[0-100]+.[0-100]+'
+    branches:
+      - 'master'
+
+jobs:
+  snyk_scan_test:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+
+      - name: Check changed Deps files
+        uses: tj-actions/changed-files@v10
+        id: verify-changed-files
+        with:
+          files: |
+            go.mod
+            ui/package-lock.json
+            website/package-lock.json
+            py/examples/requirements.txt
+            py/Makefile
+
+      - name: Snyk scan for Go dependancies
+        if: steps.verify-changed-files.outputs.any_changed == 'true'
```

Comment:
But we want to run this only when `go.mod` changed, not whenever any of our dep files changed, right?

Reply:
ok sure, I will update to make it that way. Currently, GH actions seem to have some issues as per https://www.githubstatus.com/. I will check this after this is sorted out

Reply:
Updated the scripts to run Snyk scans based on changed files only as requested